app/models/taxon_name.rb

```diff
@@ -330,6 +328,52 @@ def self.with_taxon_name_relationship(relationship)
   scope :without_otus, -> { includes(:otus).where(otus: {id: nil}) }
   scope :with_otus, -> { includes(:otus).where.not(otus: {id: nil}) }
 
+  # @return [Scope]
+  #   Combinations that are composed of children of this taxon name
+  #     when those children are not currently descendants of this taxon name
+  #
+  # !! When :cached_valid_taxon_name_id is properly set then this method is not required
+  # rather you should use :historical_taxon_names.
+  #
+  def self.out_of_scope_combinations(taxon_name_id)
+    t = ::TaxonName.arel_table
+    h = ::TaxonNameHierarchy.arel_table
+    r = ::TaxonNameRelationship.arel_table
+
+    h1 = h.alias('osch_')
+    h2 = h.alias('oschh_')
+    h3 = h.alias('oschhh_')
+
+    b = h.project(
+      # h[Arel.star],
+      # r[:subject_taxon_name_id].as('s'),
+      # r[:object_taxon_name_id].as('o'),
+      # h[:ancestor_id].as('aa'),
+      # h[:descendant_id].as('ab'),
+      # h1[:ancestor_id].as('a'),
+      h1[:descendant_id].as('b'),
+      h2[:ancestor_id].as('c'),
+      # h2[:descendant_id].as('d')
+    ).from([h])
+
+    b = b.join(r, Arel::Nodes::InnerJoin).on(h[:descendant_id].eq(r[:subject_taxon_name_id]).and(h[:ancestor_id].eq(taxon_name_id)))
```

Comment:
I suspect this could be solved with less Arel (the entire query, not just this line). Need to revisit after merge, for now if it works it is fine to leave it as-is.